packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -300,3 +372,72 @@ export class RouterliciousOrdererRestWrapper extends RouterliciousRestWrapper {
 		return restWrapper;
 	}
 }
+
+export function getW3CData(url: string) {
+	// From: https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming
+	// fetchStart: immediately before the browser starts to fetch the resource.
+	// requestStart: immediately before the browser starts requesting the resource from the server
+	// responseStart: immediately after the browser receives the first byte of the response from the server.
+	// responseEnd: immediately after the browser receives the last byte of the resource
+	//              or immediately before the transport connection is closed, whichever comes first.
+	// secureConnectionStart: immediately before the browser starts the handshake process to secure the
+	//              current connection. If a secure connection is not used, this property returns zero.
+	// startTime: Time when the resource fetch started. This value is equivalent to fetchStart.
+	// domainLookupStart: immediately before the browser starts the domain name lookup for the resource.
+	// domainLookupEnd: immediately after the browser finishes the domain name lookup for the resource.
+	// redirectStart: start time of the fetch which that initiates the redirect.
+	// redirectEnd: immediately after receiving the last byte of the response of the last redirect.
+	let dnsLookupTime: number | undefined; // domainLookupEnd - domainLookupStart
+	let redirectTime: number | undefined; // redirectEnd - redirectStart
+	let tcpHandshakeTime: number | undefined; // connectEnd  - connectStart
+	let secureConnectionTime: number | undefined; // connectEnd  - secureConnectionStart
+	let responseNetworkTime: number | undefined; // responsEnd - responseStart
+	let fetchStartToResponseEndTime: number | undefined; // responseEnd  - fetchStart
+	let reqStartToResponseEndTime: number | undefined; // responseEnd - requestStart
+	let networkTime: number | undefined; // responseEnd - responseStart
+
+	// getEntriesByType is only available in browser performance object
+	const resources1 = performance.getEntriesByType?.("resource") ?? [];
+	// Usually the latest fetch call is to the end of resources, so we start from the end.
+	for (let i = resources1.length - 1; i > 0; i--) {
+		const indResTime = resources1[i] as PerformanceResourceTiming;
+		const resource_name = indResTime.name.toString();
+		const resource_initiatortype = indResTime.initiatorType;
+		if (
+			resource_initiatortype.localeCompare("xmlhttprequest") === 0 &&
+			resource_name.includes(url)
+		) {
+			redirectTime = indResTime.redirectEnd - indResTime.redirectStart;
+			dnsLookupTime = indResTime.domainLookupEnd - indResTime.domainLookupStart;
+			tcpHandshakeTime = indResTime.connectEnd - indResTime.connectStart;
+			secureConnectionTime =
+				indResTime.secureConnectionStart > 0
```

Comment:
I believe Raja pointed to me that zero is valid value here and documentation actually says that zero mean no time was spent for this step (or something along these lines). If so, it would be better to report zero in telemetry and not undefined - it better reflects intention (and math around avg() and percentile() will be more correct)